sxt/base/type/value_type.t.cc

```diff
@@ -0,0 +1,33 @@
+/** Proofs GPU - Space and Time's cryptographic proof algorithms on the CPU and GPU.
+ *
+ * Copyright 2023-present Space and Time Labs, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#include "sxt/base/type/value_type.h"
+
+#include <type_traits>
+#include <vector>
+
+#include "sxt/base/test/unit_test.h"
+
+using namespace sxt;
+using namespace sxt::bast;
+
+TEST_CASE("we can get the value type of containers") {
+  SECTION("we handle a simple container") {
+    REQUIRE(std::is_same_v<value_type_t<std::vector<int>>, int>);
+  }
+
+  SECTION("we handle an array") { REQUIRE(std::is_same_v<value_type_t<int[6]>, int>); }
+}
```

Comment:
Should there be a section testing if a `const` type returns the appropriate value type?

Reply:
added